test/system/test_cvp_client_api.py

```diff
@@ -221,6 +221,119 @@ def test_api_user_operations(self):
         self.assertIn('total', result)
         self.assertEqual(result['total'], start_total)
 
+    def test_api_role_operations(self):
+        ''' Verify get_role, add_role and update_role
+        '''
+        # pylint: disable=too-many-statements
+        # pylint: disable=too-many-branches
+        # Test Get All Roles
+        result = self.api.get_roles()
+        self.assertIsNotNone(result)
+        self.assertIn('total', result)
+        start_total = result['total']
+
+        # Test Create Role
+        result = self.api.add_role('test_cvp_role', 'role description',
+                                    [{'name': 'image', 'mode': 'rw'}])
```

Comment:
I think this answers my question above about the type for moduleList. Likely the fix is to update the type for moduleList in the doc strings above from str to list.